src/plugins/hyperbee/hyperbee.cpp

```diff
@@ -0,0 +1,1329 @@
+    if (!set_dump_paths(fault_data->plugin->dump_dir.c_str(),
+                        dump_metadata))
+    {
+        //debug message within set_dump_paths
+        return malware;
+    }
+
+    //Is used to find a checksum in the set of already analyzed memory parts
+    auto memory_hash_identifier = fault_data->plugin->dumped_memory_map.find(dump_metadata->sha256sum);
+
+    //If the checksum already exists:
+    if (memory_hash_identifier != fault_data->plugin->dumped_memory_map.end())
+    {
+        //Make sure the two file stems match
+        if (strcmp(memory_hash_identifier->second.c_str(), dump_metadata->file_stem) == 0)
+        {
+            //Increase the dump counter in this special case
+            ++fault_data->plugin->dump_id;
+            return malware;
+        }
+        else
+        {
+            //since there were problems with the file stem, continue to dump all data again
+            PRINT_DEBUG("[HYPERBEE] Could not reuse data from previously saved data, dumping again\n");
+        }
+
+        //If the flow continues here, there was an or with the stored data. Dump the data again.
+    }
+
+    //At this point I use a yet unpublished tool from Politecnico di Milano (IT) to analyse the accessed memory page for API calls. This information is used within a malware classifier.
+    //TODO FutureWork  malware = malware_classifier(...)
+
+    //If malware was detected or manually switched to always dump
+    if (malware)
+    {
+        // Comment from memdump.cpp:
+        // The file name format for the memory dump file is:
+        // <dump base address>_<16 chars of hash>
+        // This was set in order to satisfy the following issue:
+        // * when disassembling, it is required to know the dump's image base, here it could be obtained
+        //   just by looking at the file name which is handy both for humans and automated processing
+        // * no other information was included in the file name to make it possible to reference this file in the
+        //   future if the identical memory would be dumped again
+
+        //If the dump fails
+        if (!dump_memory_region(vmi,
+                                fault_data->plugin,
+                                &ctx_memory_dump,
+                                dump_metadata))
+        {
+            PRINT_DEBUG("[HYPERBEE] Could not dump memory\n");
+            if (dump_metadata->dump_file)
+            {
+                g_free(dump_metadata->dump_file);
+                dump_metadata->dump_file = nullptr;
+            }
+            return malware;
+        }
+
+        //If the dump was successful: increase the dump counter
+        ++fault_data->plugin->dump_id;
+
+        //Add the checksum as key with the file_name_prefix (as data) to the map.
+        fault_data->plugin->dumped_memory_map.insert(std::pair<std::string,
+                std::string>(dump_metadata->sha256sum,
+                             dump_metadata->file_stem));
```

Comment:
```suggestion
        fault_data->plugin->dumped_memory_map.insert(std::make_pair(dump_metadata->sha256sum, dump_metadata->file_stem));
```
There is a utility for this, so you don't have to specify types :)

Reply:
Thanks, changed as well!